src/Aspire.Dashboard/Model/ResourceSourceViewModel.cs

```diff
@@ -31,46 +34,52 @@ public class ResourceSourceViewModel(string value, List<LaunchArgument>? content
                         .Select((arg, i) => new LaunchArgument(arg, IsShown: !areArgumentsSensitive[i]))
                         .ToList();
 
-                    commandLineInfo = (Arguments: arguments, argumentsString);
+                    commandLineInfo = new CommandLineInfo(
+                        Arguments: arguments,
+                        ArgumentsString: argumentsString,
+                        TooltipString: string.Join(" ", arguments.Select(arg => arg.IsShown
+                            ? arg.Value
+                            : WebUtility.HtmlDecode(DashboardUIHelpers.GetMaskingText(6).Value))));
```

Comment:
Add a unit test to ResourceSourceViewModel tests that tests this.

This PR could be backported so it should be well tested.

Reply:
there is a case in ResourceSourceViewModel_ReturnsCorrectValue_TestData that tests this specific case!